### ISSUE TITLE
Discrepency between total time and execution time

### ISSUE THREAD

**Issue by goldmanm** [ReactionMechanismGenerator/RMG-Py#1326]
I found a potentially large discrepency within the RMG.log file regarding timing.

From the RMG.log file, the first and last line give an indication of how long a run takes:

```
RMG execution initiated at Fri Mar  9 14:58:55 2018
RMG execution terminated at Fri Mar  9 21:48:36 2018
```

In the same log file, the last mention of execution time is given as `Execution time (DD:HH:MM:SS): 00:03:26:15`. 

From the first timing, it seems like the job took 6 hours and 50 minutes. However, the execution time section only asserts 3 hours 26 minutes. This seems like a huge discrepancy to me. 

The timing isn't really proportional either. Another job started at the same time took 6 hours 34 minutes according to method 1 and 4 hours 32 minutes according the execution time statement. 

Has anyone else encountered this? Do we know why this is happening?

**Comment by mliu49** [comment]
I think it's just due to when the `Execution time` is recorded. That's done at the end of each iteration, and there are a number of things RMG does to wrap up the job after the last iteration is finished.

However, the fact that the discrepancy is so large is definitely cause for concern.

Major tasks that are done between the last `Execution time` and the true end of the job:
- Seed mech generation
- Sensitivity analysis
- Cantera file generation
- Model checking

I know we previously found seed mech generation and model checking to consume inordinate amounts of time, but they should have been trimmed down significantly. Were the examples you cited on master?

The improved profiling branch also separately profiles the post-final-iteration part of the job, if you would like to investigate further. I didn't include processing/graph generation for that data, but there's a summary in the `profile.log` file, and the raw data is stored in `post.profile`.

**Comment by goldmanm** [comment]
The branch used was master updated on March 9th, and I did not run any sensitivity analysis.

It also depends on when the models starts checking time, which could be after building the database. I am not sure.

It just seemed like a huge discrepancy. Even 15 minute difference would not raise this much alarm.

**Comment by mliu49** [comment]
I think that makes sense. We merged #1317 last week, which removed the reaction duplicate checking implemented by #1290 at the beginning of May. That step was taking a lot of time because it was comparing every edge reaction to each other.
